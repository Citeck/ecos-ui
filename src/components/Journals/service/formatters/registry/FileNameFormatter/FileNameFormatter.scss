.ecos-filename-formatter {
  display: grid;
  grid-auto-flow: column;
  align-items: start;
  justify-content: center;
  grid-gap: 10px;
  height: 100%;

  &__icon {
    height: 100%;
    width: 1.6em;
  }
}
