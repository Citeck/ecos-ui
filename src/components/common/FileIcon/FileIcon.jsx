import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import formats from './font/icons/high-contrast/catalog';
import './font/file-icon-high-contrast.css';
import './FileIcon.scss';

const FileIcon = ({ format, className }) => {
  let isCorrectFormat = false;
  let lowerCaseFormat = format || '';
  if (format && typeof format.toLowerCase === 'function') {
    lowerCaseFormat = format.toLowerCase();
    isCorrectFormat = Array.from(formats).includes(lowerCaseFormat);
  }

  return (
    <span
      className={classNames('ecos-file-icon', 'fiv-hct', className, {
        [`fiv-icon-${lowerCaseFormat}`]: isCorrectFormat,
        'fiv-icon-blank': !isCorrectFormat || lowerCaseFormat === 'blank'
      })}
    />
  );
};

FileIcon.propTypes = {
  className: PropTypes.string,
  format: PropTypes.oneOf(Array.from(formats))
};

export default FileIcon;
