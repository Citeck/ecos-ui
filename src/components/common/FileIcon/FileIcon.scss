.ecos-file-icon {
  &:before {
    content: '';
  }
}
