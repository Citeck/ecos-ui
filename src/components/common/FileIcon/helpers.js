import formats from './font/icons/high-contrast/catalog';
import { blankFormat } from './constants';

export function detectFormat(fileName) {
  try {
    if (typeof fileName !== 'string') {
      return blankFormat;
    }

    const parts = fileName.split('.');
    if (parts.length < 2) {
      return blankFormat;
    }

    const lastPart = parts[parts.length - 1].toLowerCase();
    if (Array.from(formats).includes(lastPart)) {
      return lastPart;
    }

    return blankFormat;
  } catch (e) {
    return blankFormat;
  }
}
