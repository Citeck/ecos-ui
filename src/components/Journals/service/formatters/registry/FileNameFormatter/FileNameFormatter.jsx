import React from 'react';

import { REMOTE_TITLE_ATTR_NAME } from '../../../../../../constants/pageTabs';
import FileIcon from '../../../../../common/FileIcon';
import { detectFormat } from '../../../../../common/FileIcon/helpers';

import BaseFormatter from '../BaseFormatter';
import { URL } from '../../../../../../constants';

import './FileNameFormatter.scss';

export default class FileNameFormatter extends BaseFormatter {
  static TYPE = 'filename';

  /**
   * @param {FormatterProps} props
   * @return {React.ReactNode}
   */
  format(props = {}) {
    const { row = {}, cell } = props;

    const url = `${URL.DASHBOARD}?recordRef=${row.id}`;
    const linkProps = {
      target: '_blank',
      rel: 'noopener noreferrer',
      ...{ [REMOTE_TITLE_ATTR_NAME]: true }
    };

    return (
      <a href={url} {...linkProps} className="ecos-filename-formatter">
        <FileIcon format={detectFormat(cell)} className="ecos-filename-formatter__icon" />
        {cell}
      </a>
    );
  }
}
