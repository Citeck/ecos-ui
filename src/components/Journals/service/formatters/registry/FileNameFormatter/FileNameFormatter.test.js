import { render } from '@testing-library/react';

import FileIcon from '../../../../../common/FileIcon';

import FileNameFormatter from './FileNameFormatter';

const fileNameFormatterInstance = new FileNameFormatter();

describe('FileNameFormatter', () => {
  it('getType should return correct type', () => {
    expect(fileNameFormatterInstance.getType()).toBe(FileNameFormatter.TYPE);
  });
  describe('format method', () => {
    it('should render file icon', () => {
      const { container } = render(fileNameFormatterInstance.format());
      expect(container.getElementsByClassName('fiv-icon-blank')).toHaveLength(1);
      expect(container).toMatchSnapshot();
    });
  });
});
