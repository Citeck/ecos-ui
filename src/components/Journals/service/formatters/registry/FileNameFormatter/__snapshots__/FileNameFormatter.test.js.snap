// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileNameFormatter format method should render file icon 1`] = `
<div>
  <a
    class="ecos-filename-formatter"
    data-remote-title="true"
    href="/v2/dashboard?recordRef=undefined"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="ecos-file-icon fiv-hct ecos-filename-formatter__icon fiv-icon-blank"
    />
  </a>
</div>
`;
